### ISSUE TITLE
issues in mzTab when description of protein contains []

### ISSUE THREAD

**Issue by ypriverol** [levitsky/pyteomics#3]
@levitsky: 

We are using pyteomics for some of our pipelines. I have seen that when the description of the protein contains `[ ]` the current mzTab fail reading the description part. 

```
['PRT', 'sp|Q15119|PDK2_HUMAN', '[Pyruvate dehydrogenase (acetyl-transferring)] kinase isozyme 2, mitochondrial OS=Homo sapiens OX=9606 GN=PDK2 PE=1 SV=2', 'null', 'null', 'uniprot-homo-sapiens_decoy.fasta', 'null', 'null', '0.038215846688334', 'sp|Q15119|PDK2_HUMAN', '33-UNIMOD:35', '0.208845208845209', '9.72674e06', '8.77109875e05', '0.0', '0.0', '0.0', '7.632018e06', '0.0', '0.0', '0.0', '9.72674e06', 'null', 'null', '8.77109875e05', 'null', 'null', '0.0', 'null', 'null', '0.0', 'null', 'null', '0.0', 'null', 'null', '7.632018e06', 'null', 'null', '0.0', 'null', 'null', '0.0', 'null', 'null', '0.0', 'null', 'null', 'indistinguishable_proteins']
``` 
it fails in the following line: 

```python
cv, acc, name, value = re.split(r"\s*,\s*", tuplet[1:-1]) 
```

**Comment by levitsky** [comment]
Hi Yasset,

I am not sure that the code you are showing applies to that line. Would you be able to share a minimal example of a problematic file (or not minimal, depending on how sensitive the data are)?

**Comment by mobiusklein** [comment]
I think I see the failure path. In `_cast_value` https://github.com/levitsky/pyteomics/blob/master/pyteomics/mztab.py#L81-L82, there is an indiscriminant decision that if something starts with a "[" it must be a param group. The solution is to try to parse the value  as a parameter and if we fail, return the string unparsed, or somehow inject the name of the column being parsed and know whether that column can be a param in that context. Easier to do the former for now.

**Comment by levitsky** [comment]
@ypriverol 
This issue got auto-closed by merging #4, but please drop a comment on whether it's actually fixed for you now.

@mobiusklein 
Thank you for stepping in!
